core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -387,6 +362,31 @@ private[deploy] class Master(
   }
 
   override def receiveAndReply(context: RpcCallContext): PartialFunction[Any, Unit] = {
+    case RegisterWorker(
+        id, workerHost, workerPort, workerRef, cores, memory, workerUiPort, publicAddress) => {
+      logInfo("Registering worker %s:%d with %d cores, %s RAM".format(
+        workerHost, workerPort, cores, Utils.megabytesToString(memory)))
+      if (state == RecoveryState.STANDBY) {
+        // ignore, don't send response
```

Comment:
wait, actually that's a problem. Right now we have a `send` and an `ask`. The `send` actually won't be received by anyone because we only handle this in `receiveAndReply`